contracts/receivables/RealWorldReceivable.sol

```diff
@@ -0,0 +1,198 @@
+        address paymentToken,
+        uint96 receivableAmount,
+        uint64 maturityDate,
+        string memory uri
+    ) public onlyRole(MINTER_ROLE) {
+        uint256 tokenId = _tokenIdCounter.current();
+        _tokenIdCounter.increment();
+        _safeMint(recipient, tokenId);
+
+        ReceivableInfo memory receivableInfo = ReceivableInfo(
+            paymentAddress,
+            paymentToken,
+            receivableAmount,
+            0, // paidAmount
+            maturityDate
+        );
+        receivableInfoMapping[tokenId] = receivableInfo;
+
+        _setTokenURI(tokenId, uri);
+    }
+
+    /**
+     * @dev Declares payment for a receivable.
+     * Only the owner of the token can declare a payment.
+     * The payment method for the receivable must be Declarative.
+     * The receivable must not already be paid in full.
+     * Emits a `PaymentDeclared` event.
+     * @param tokenId The ID of the receivable token.
+     * @param paymentAmount The amount of payment being declared.
+     */
+    function declarePayment(uint256 tokenId, uint96 paymentAmount) public {
+        if (msg.sender != ownerOf(tokenId)) revert Errors.notNFTOwner();
+        ReceivableInfo storage receivableInfo = receivableInfoMapping[tokenId];
+
+        if (receivableInfo.paidAmount >= receivableInfo.receivableAmount)
+            revert Errors.receivableAlreadyPaid();
+
+        receivableInfo.paidAmount += uint96(paymentAmount);
+
+        emit PaymentDeclared(
+            msg.sender,
+            receivableInfo.poolAddress,
+            tokenId,
+            uint256(paymentAmount)
+        );
+    }
+
+    /**
+     * @dev Gets the payment status of a receivable.
+     * Returns `Status.Paid` if the receivable has been paid in full.
+     * Returns `Status.PartiallyPaid` if the receivable has been paid partially.
+     * Returns `Status.Unpaid` if the receivable has not been paid at all.
+     * @param tokenId The ID of the receivable token.
+     * @return The payment status of the receivable.
+     */
+    function getPaymentStatus(uint256 tokenId) public view returns (Status) {
+        ReceivableInfo storage receivableInfo = receivableInfoMapping[tokenId];
+        if (receivableInfo.paidAmount == receivableInfo.receivableAmount) {
+            return Status.Paid;
+        } else if (receivableInfo.paidAmount > 0) {
+            return Status.PartiallyPaid;
+        } else {
+            return Status.Unpaid;
+        }
+    }
+
+    // The following functions are overrides required by Solidity.
+    function _beforeTokenTransfer(
+        address from,
+        address to,
+        uint256 tokenId,
+        uint256 batchSize
+    ) internal override(ERC721Upgradeable, ERC721EnumerableUpgradeable) {
+        super._beforeTokenTransfer(from, to, tokenId, batchSize);
+    }
+
+    function _burn(uint256 tokenId)
+        internal
+        override(ERC721Upgradeable, ERC721URIStorageUpgradeable)
+    {
+        super._burn(tokenId);
```

Comment:
What was the original comment/concern?

Reply:
Bin suggested we should explicitly define which class to call (i.e. ERC721URIStorageUpgradeable._burn instead of super._burn), but as the link explains this is actually unnecessary.

Reply:
Sounds good. Do you mind adding a line of comment to show super inheritance is from right to left to reduce a layer of cognitive load for the readers?

Reply:
Added.